### PR TITLE
Feature/591

### PR DESCRIPTION
**Pull request by au2**
If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### How should this be manually tested?

1) A new field (column) "type" is added to surveys to differentiate feedback surveys from others.
2) A feedback survey can be assigned as default or survey specific feedback surveys using 'feedback-surveys' and 'default-feedback-survey' end point by admin.
3) Client can get/show/answer a feedback survey for a specific survey.
4) Survey type can be changed only when survey has never been answered or no feedback assignment is done.
5) user-surveys end point filters feedback surveys so they are only available to participant when client chooses to show them.

### CODE REVIEW

File: models/db/feedback-survey.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            type: Sequelize.INTEGER,
+            field: 'survey_id',
+            references: {
+                model: {
+                    schema,
+                    tableName: 'survey',
+                },
+                key: 'id',
+            },
+        },
```
Comment (unprofessional) [inline models/db/feedback-survey.js]: Why have two local columns both reference the same remote table.column?

Comment (unprofessional) [reply models/db/feedback-survey.js]: Oh. I get it. Resuming API testing.

**Comment by unprofessional** [comment]
The swagger docs indicate that a successful POST to `/feedback-surveys` and `/default-feedback-survey` should return an entry `id` but performing those actions, while ultimately successful, returns nothing.

I wouldn't expect it to, though, really, since I don't think there's much use in getting association ids in this scenario.

**Comment by au2** [comment]
corrected

File: models/db/feedback-survey.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (au2) [reply models/db/feedback-survey.js]: corrected

**Comment by unprofessional** [comment]
Everything else looks good to me. 👍

**Comment by au2** [comment]
Can I move this now or do we need a second reviewer now?

**Comment by unprofessional** [comment]
Looks like we need a second reviewer on all things from now on. So assigning @sirctseb .

**Comment by sirctseb** [comment]
Are we concerned that this drifts rr surveys away from the survey service? Will this introduce breaking changes into indaba if we port it into survey service?

**Comment by au2** [comment]
No, this does not change anything for indaba if they do not create any survey of type feedback. If they introduce some feedback functionality they can possibly use this as well.

**Pull request merged**